Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -262,6 +262,7 @@ open class NavigationMapView: UIView {
                 self.mapView.location.options.puckType = .puck3D(configuration)
             }
         }
+        mapView.location.options.puckBearingSource = .course
```

Comment:
Right now the map puck bearing source defaults to the location course, while the `simulatesLocation` is only used to notify the `navigationMapView` that we're using simulated location provider, and keep the puck synced with vanishing route line synced with user location layer.